letsencrypt-apache/letsencrypt_apache/configurator.py

```diff
@@ -966,6 +967,66 @@ def enhance(self, domain, enhancement, options=None):
             logger.warn("Failed %s for %s", enhancement, domain)
             raise
 
+    def _enable_ocsp_stapling(self, ssl_vhost, unused_options):
+        """Enables OCSP Stapling
+
+        In OCSP, each client (e.g. browser) would have to query the
+        OCSP Responder to validate that the site certificate was not revoked.
+
+        Enabling OCSP Stapling, would allow the web-server to query the OCSP
+        Responder, and staple its response to the offered certificate during
+        TLS. i.e. clients would not have to query the OCSP responder.
+
+        .. note:: This function saves the configuration
+
+        :param ssl_vhost: Destination of traffic, an ssl enabled vhost
+        :type ssl_vhost: :class:`~letsencrypt_apache.obj.VirtualHost`
+
+        :param unused_options: Not currently used
+        :type unused_options: Not Available
+
+        :returns: Success, general_vhost (HTTP vhost)
+        :rtype: (bool, :class:`~letsencrypt_apache.obj.VirtualHost`)
+
+        """
+        min_apache_ver = (2, 4) #TODO check min apache ver that supports stapling
+        if self.version >= min_apache_ver:
+            if "socache_shmcb_module" not in self.parser.modules:
+                self.enable_mod("socache_shmcb")
+
+            # Check if there's an existing SSLUseStapling directive on.
+            use_stapling_aug_path = self.parser.find_dir("SSLUseStapling",
+                    "on", start=ssl_vhost.path)
+            if not use_stapling_aug_path:
+                self.parser.add_dir(ssl_vhost.path, "SSLUseStapling", "on")
+
+            ssl_vhost_aug_path = parser.get_aug_path(ssl_vhost.filep)
+
+            # Check if there's an existing SSLStaplingCache directive.
+            stapling_cache_aug_path = self.parser.find_dir('SSLStaplingCache',
+                    None, ssl_vhost_aug_path)
+
+            # We'll simply delete the directive, as it might be something like
+            # SSLStaplingCache /tmp/ocsp_stapling
+            # The OS usually cleans /tmp on reboot, which means that the server
```

Comment:
Note that this is not really a big problem - if a server re-requests the OCSP response on reboot, it's not a big deal. Still, it's nice to have a consistent cache path.